Add `lxd` group creation to installation process (#13382)
canonical/lxd: doc/installing.md

echo "root:1000000:1000000000" | sudo tee -a /etc/subuid /etc/subgid
```

By default, only users added to the `lxd` group can interact with the LXD daemon. Installing from source doesn't guarantee that the `lxd` group exists in the system. If you want the current user (or any other user) to be able to interact with the LXD daemon, add it to the `lxd` group:

```bash

> grep -q "^lxd:" /etc/group && sudo groupadd lxd # create the group if needed
> grep -q "^lxd:" /etc/group | grep -q $USER && sudo usermod -aG lxd $USER

Reviewer: ```suggestion
getent group lxd >/dev/null || sudo groupadd --system lxd # create the group if needed
getent group lxd | grep -qwF "$USER" || sudo usermod -aG lxd "$USER"
```

The root user and all members of the `lxd` group can interact with the local daemon.
See {ref}`security-daemon-access` for more information.

> On Ubuntu images, the `lxd` group is already created and the main user is automatically added to it. The group is also created during installation if you {ref}`installed LXD from the snap<installing-from-package>`. If the `lxd` group is missing on your system(as might be the case if you {ref}`installed LXD from the sources <installing-from-source>`), create it and restart the LXD daemon:

Reviewer 2: ```suggestion
On Ubuntu images, the `lxd` group already exists and the main user is automatically added to it. The group is also created during installation if you {ref}`installed LXD from the snap<installing-from-package>`. If the `lxd` group is missing on your system (as might be the case if you {ref}`installed LXD from the sources <installing-from-source>`), create it and restart the LXD daemon:
```

```bash
grep -q "^lxd:" /etc/group && sudo groupadd lxd
```

> No users are added to the group on installation. You can then add trusted users to the group.

Reviewer 2: ```suggestion
No users are added to the group on installation. You must add trusted users to the group so they can use LXD:
```

The root user and all members of the `lxd` group can interact with the local daemon.
See {ref}`security-daemon-access` for more information.

On Ubuntu images, the `lxd` group already exists and the main user is automatically added to it. The group is also created during installation if you {ref}`installed LXD from the snap<installing-from-package>`. If the `lxd` group is missing on your system (as might be the case if you {ref}`installed LXD from the sources <installing-from-source>`), create it and restart the LXD daemon:

```bash

> grep -q "^lxd:" /etc/group && sudo groupadd lxd

Reviewer: ```suggestion
getent group lxd >/dev/null || sudo groupadd --system lxd
```

```

No users are added to the group on installation. You must add trusted users to the group so they can use LXD:

```bash
grep -q "^lxd:" /etc/group | grep -q $USER && sudo usermod -aG lxd $USER # adding current user as an example
```

> Anyone added to this group will have full control over LXD.

Reviewer: I think we should probably link to [doc/explanation/security.md](https://documentation.ubuntu.com/lxd/en/latest/explanation/security/#local-access-to-the-lxd-daemon) as being a member of that group is akin to being root.
